core/templates/pages/landing-pages/topic-landing-page/topic-landing-page.component.ts

```diff
@@ -17,118 +17,146 @@
-        }, 150);
-      };
-
-      ctrl.$onInit = function() {
-        var pathArray = WindowRef.nativeWindow.location.pathname.split('/');
-        var subjectName = pathArray[1];
-        var topicName = pathArray[2];
-        topicData = TOPIC_LANDING_PAGE_DATA[subjectName][topicName];
-        ctrl.topicTitle = topicData.topicTitle;
-
-        ctrl.lessonsQualities = getLessonQualities();
-        ctrl.backgroundBannerUrl = (
-          UrlInterpolationService.getStaticImageUrl(
-            '/background/bannerB.svg'));
-
-        var topicImageUrlTemplate = '/landing/<subject>/<topic>/<filename>';
-        ctrl.lessonInDevicesPngImageSrc = (
-          UrlInterpolationService.getStaticImageUrl(
-            UrlInterpolationService.interpolateUrl(
-              topicImageUrlTemplate, {
-                subject: subjectName,
-                topic: topicName,
-                filename: 'lesson_in_devices.png'
-              })));
-        ctrl.lessonInDevicesWebpImageSrc = (
-          UrlInterpolationService.getStaticImageUrl(
-            UrlInterpolationService.interpolateUrl(
-              topicImageUrlTemplate, {
-                subject: subjectName,
-                topic: topicName,
-                filename: 'lesson_in_devices.webp'
-              })));
-
-        PageTitleService.setPageTitle(
-          [ctrl.topicTitle, topicData.topicTagline, 'Oppia'].join(' | '));
-      };
-    }
-  ]
-});
+
+import { Component, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+import { PageTitleService } from 'services/page-title.service';
+import { TopicLandingPageConstants } from
+  'pages/landing-pages/topic-landing-page/topic-landing-page.constants';
+
+
+interface ILessonsQuality {
+  title: string;
+  description: string;
+  imagePngFilename: string;
+  imageWebpFilename: string;
+  imageAlt: string;
+}
+
+interface ITopicData {
+  topicTitle: string,
+  topicTagline: string,
+  collectionId: string,
+  chapters: string[]
+}
+
+@Component({
+  selector: 'topic-landing-page',
+  templateUrl: './topic-landing-page.component.html',
+  styleUrls: []
+})
+export class TopicLandingPageComponent implements OnInit {
+  backgroundBannerUrl: string = null;
+  lessonInDevicesPngImageSrc: string = null;
+  lessonInDevicesWebpImageSrc: string = null;
+  lessonsQualities: ILessonsQuality[] = null;
+  topicData: ITopicData = null;
+  topicTitle: string = null;
+
+  constructor(
+    private pageTitleService: PageTitleService,
+    private siteAnalyticsService: SiteAnalyticsService,
+    private urlInterpolationService: UrlInterpolationService,
+    private windowRef: WindowRef) {}
+
+  getLessonQualities(): ILessonsQuality[] {
```

Comment:
Use notation `Array<LessonsQuality>`

Reply:
Is this a recent style change? I remember @nishantwrp had a different opinion https://github.com/oppia/oppia/pull/9570#discussion_r442875384

Changed it for now

Reply:
No, it has been decided that interfaces won't be prefixed with "I". Would it be possible for you to keep a note of all these changes in style? We can add it to the guide.

Reply:
Okay, but I was talking about Array<type> vs type[]

Reply:
Yeah, I'll keep a note of the styling changes

Reply:
> Okay, but I was talking about Array vs type[]

Oh, let me confirm, don't change this then.

Reply:
type[] is the way go!